modules/currency-l0/src/main/scala/org/tessellation/currency/l0/snapshot/programs/Rollback.scala

```diff
@@ -70,7 +71,20 @@ object Rollback {
 
       _ <- dataApplication.map {
         case ((da, cs)) =>
-          val fetchSnapshot: Hash => F[Option[Hashed[GlobalIncrementalSnapshot]]] = (hash: Hash) => globalL0Service.pullGlobalSnapshot(hash)
+          val attempts_to_fetch_global_snapshot = 10
+          val fetchSnapshot: Hash => F[Option[Hashed[GlobalIncrementalSnapshot]]] = (hash: Hash) => {
+            def attemptFetch(remainingAttempts: Int): F[Option[Hashed[GlobalIncrementalSnapshot]]] =
+              globalL0Service.pullGlobalSnapshot(hash).flatMap {
+                case Some(snapshot) => snapshot.some.pure
+                case None if remainingAttempts > 1 =>
+                  logger.warn("Could not get global snapshot, trying again with another peer") >> attemptFetch(remainingAttempts - 1)
+                case None =>
+                  logger.error("Could not get global snapshot after 10 tries") >>
```

Comment:
message is not correlated with the variable, so it's easy to make a mistake (changing one place but not the other)